{
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "@scoder this looks like the same bug as #2199 . It occurs on OSX with clang using cython 0.28.4:\r\n\r\n```\r\n* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=EXC_I386_GPFLT)\r\n  * frame #0: 0x00000001000e4608 libpython2.7.dylib`_PyObject_GetDictPtr + 8\r\n    frame #1: 0x0000000103d1e27b Code.so`__Pyx_AddTraceback + 107\r\n    frame #2: 0x0000000103d8b234 Code.so`__pyx_pw_6Cython_8Compiler_4Code_11GlobalState_61lookup_filename + 1924\r\n    frame #3: 0x0000000103d3b80b Code.so`__pyx_pw_6Cython_8Compiler_4Code_11CCodeWriter_205lookup_filename + 379\r\n    frame #4: 0x0000000103d3df31 Code.so`__pyx_pw_6Cython_8Compiler_4Code_11CCodeWriter_195error_goto + 1137\r\n    frame #5: 0x0000000103d3d387 Code.so`__pyx_pw_6Cython_8Compiler_4Code_11CCodeWriter_197error_goto_if + 759\r\n    frame #6: 0x000000010009dc55 libpython2.7.dylib`PyObject_Call + 101\r\n    frame #7: 0x000000010014a168 libpython2.7.dylib`PyEval_EvalFrameEx + 22088\r\n    frame #8: 0x00000001001448c4 libpython2.7.dylib`PyEval_EvalCodeEx + 2164\r\n    frame #9: 0x000000010014ec0d libpython2.7.dylib`fast_function + 109\r\n```",
    "closed_at": null,
    "comment_data": [
        {
            "author_association": "MEMBER",
            "body": "Could you please provide the complete stack trace? I double checked the code here and fail to see a way how this can occur with 0.28.4.",
            "created_at": "2018-07-21T18:31:55Z",
            "html_url": "https://github.com/cython/cython/issues/2505#issuecomment-406814914",
            "id": 406814914,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/2505",
            "node_id": "MDEyOklzc3VlQ29tbWVudDQwNjgxNDkxNA==",
            "updated_at": "2018-07-21T18:31:55Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/406814914",
            "user": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "node_id": "MDQ6VXNlcjQ5MTY1OQ==",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            }
        },
        {
            "author_association": "NONE",
            "body": "okay I checked on TRAVIS the OSX version of Cython\r\n```\r\nTraviss-Mac-463:tardis travis$ source activate tardis\r\n(tardis) Traviss-Mac-463:tardis travis$ python\r\nPython 2.7.15 |Anaconda, Inc.| (default, May  1 2018, 18:37:05) \r\n[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin\r\nType \"help\", \"copyright\", \"credits\" or \"license\" for more information.\r\n>>> import cython\r\n>>> cython.__path__\r\nTraceback (most recent call last):\r\n  File \"<stdin>\", line 1, in <module>\r\nAttributeError: 'module' object has no attribute '__path__'\r\n>>> cython.__version__\r\n'0.28.4'\r\n>>> import Cython\r\n>>> Cython.__version__\r\n'0.28.4'\r\n```",
            "created_at": "2018-07-23T10:15:06Z",
            "html_url": "https://github.com/cython/cython/issues/2505#issuecomment-407009053",
            "id": 407009053,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/2505",
            "node_id": "MDEyOklzc3VlQ29tbWVudDQwNzAwOTA1Mw==",
            "updated_at": "2018-07-23T10:15:06Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/407009053",
            "user": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/506116?v=4",
                "events_url": "https://api.github.com/users/wkerzendorf/events{/privacy}",
                "followers_url": "https://api.github.com/users/wkerzendorf/followers",
                "following_url": "https://api.github.com/users/wkerzendorf/following{/other_user}",
                "gists_url": "https://api.github.com/users/wkerzendorf/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/wkerzendorf",
                "id": 506116,
                "login": "wkerzendorf",
                "node_id": "MDQ6VXNlcjUwNjExNg==",
                "organizations_url": "https://api.github.com/users/wkerzendorf/orgs",
                "received_events_url": "https://api.github.com/users/wkerzendorf/received_events",
                "repos_url": "https://api.github.com/users/wkerzendorf/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/wkerzendorf/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/wkerzendorf/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/wkerzendorf"
            }
        },
        {
            "author_association": "NONE",
            "body": "This is starting up lldb\r\n```\r\n(tardis) Traviss-Mac-463:tardis travis$ lldb python\r\n(lldb) target create \"python\"\r\nTraceback (most recent call last):\r\n  File \"<input>\", line 1, in <module>\r\n  File \"/Users/travis/miniconda/envs/tardis/lib/python2.7/copy.py\", line 52, in <module>\r\n    import weakref\r\n  File \"/Users/travis/miniconda/envs/tardis/lib/python2.7/weakref.py\", line 14, in <module>\r\n    from _weakref import (\r\nImportError: cannot import name _remove_dead_weakref\r\nCurrent executable set to 'python' (x86_64).\r\n```",
            "created_at": "2018-07-23T10:15:57Z",
            "html_url": "https://github.com/cython/cython/issues/2505#issuecomment-407009261",
            "id": 407009261,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/2505",
            "node_id": "MDEyOklzc3VlQ29tbWVudDQwNzAwOTI2MQ==",
            "updated_at": "2018-07-23T10:15:57Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/407009261",
            "user": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/506116?v=4",
                "events_url": "https://api.github.com/users/wkerzendorf/events{/privacy}",
                "followers_url": "https://api.github.com/users/wkerzendorf/followers",
                "following_url": "https://api.github.com/users/wkerzendorf/following{/other_user}",
                "gists_url": "https://api.github.com/users/wkerzendorf/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/wkerzendorf",
                "id": 506116,
                "login": "wkerzendorf",
                "node_id": "MDQ6VXNlcjUwNjExNg==",
                "organizations_url": "https://api.github.com/users/wkerzendorf/orgs",
                "received_events_url": "https://api.github.com/users/wkerzendorf/received_events",
                "repos_url": "https://api.github.com/users/wkerzendorf/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/wkerzendorf/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/wkerzendorf/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/wkerzendorf"
            }
        },
        {
            "author_association": "NONE",
            "body": "@scoder - maybe the version 0.28.4 in the conda-forge osx channel is not 0.28.4 - who is the maintainer of that?",
            "created_at": "2018-07-23T10:20:35Z",
            "html_url": "https://github.com/cython/cython/issues/2505#issuecomment-407010596",
            "id": 407010596,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/2505",
            "node_id": "MDEyOklzc3VlQ29tbWVudDQwNzAxMDU5Ng==",
            "updated_at": "2018-07-23T10:20:35Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/407010596",
            "user": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/506116?v=4",
                "events_url": "https://api.github.com/users/wkerzendorf/events{/privacy}",
                "followers_url": "https://api.github.com/users/wkerzendorf/followers",
                "following_url": "https://api.github.com/users/wkerzendorf/following{/other_user}",
                "gists_url": "https://api.github.com/users/wkerzendorf/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/wkerzendorf",
                "id": 506116,
                "login": "wkerzendorf",
                "node_id": "MDQ6VXNlcjUwNjExNg==",
                "organizations_url": "https://api.github.com/users/wkerzendorf/orgs",
                "received_events_url": "https://api.github.com/users/wkerzendorf/received_events",
                "repos_url": "https://api.github.com/users/wkerzendorf/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/wkerzendorf/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/wkerzendorf/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/wkerzendorf"
            }
        },
        {
            "author_association": "NONE",
            "body": "@scoder - now I can't even get cython to segfault when running in LLDB in the TRAVIS debug mode. It still segfaults in the Travis build. We will disable OSX builds for now in TRAVIS and see if the next release fixes the issue. I wonder if cython 0.28.4 in OSX-build on conda-forge is maybe not the code version 0.28.4 but an earlier version that is falsely labeled. ",
            "created_at": "2018-07-24T09:39:22Z",
            "html_url": "https://github.com/cython/cython/issues/2505#issuecomment-407345783",
            "id": 407345783,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/2505",
            "node_id": "MDEyOklzc3VlQ29tbWVudDQwNzM0NTc4Mw==",
            "updated_at": "2018-07-24T09:39:22Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/407345783",
            "user": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/506116?v=4",
                "events_url": "https://api.github.com/users/wkerzendorf/events{/privacy}",
                "followers_url": "https://api.github.com/users/wkerzendorf/followers",
                "following_url": "https://api.github.com/users/wkerzendorf/following{/other_user}",
                "gists_url": "https://api.github.com/users/wkerzendorf/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/wkerzendorf",
                "id": 506116,
                "login": "wkerzendorf",
                "node_id": "MDQ6VXNlcjUwNjExNg==",
                "organizations_url": "https://api.github.com/users/wkerzendorf/orgs",
                "received_events_url": "https://api.github.com/users/wkerzendorf/received_events",
                "repos_url": "https://api.github.com/users/wkerzendorf/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/wkerzendorf/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/wkerzendorf/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/wkerzendorf"
            }
        }
    ],
    "comments": 5,
    "comments_url": "https://api.github.com/repos/cython/cython/issues/2505/comments",
    "created_at": "2018-07-18T18:02:56Z",
    "event_data": [
        {
            "actor": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "node_id": "MDQ6VXNlcjQ5MTY1OQ==",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2018-07-18T18:02:56Z",
            "event": "mentioned",
            "id": 1740782446,
            "node_id": "MDE0Ok1lbnRpb25lZEV2ZW50MTc0MDc4MjQ0Ng==",
            "url": "https://api.github.com/repos/cython/cython/issues/events/1740782446"
        },
        {
            "actor": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "node_id": "MDQ6VXNlcjQ5MTY1OQ==",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2018-07-18T18:02:56Z",
            "event": "subscribed",
            "id": 1740782447,
            "node_id": "MDE1OlN1YnNjcmliZWRFdmVudDE3NDA3ODI0NDc=",
            "url": "https://api.github.com/repos/cython/cython/issues/events/1740782447"
        },
        {
            "actor": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "node_id": "MDQ6VXNlcjQ5MTY1OQ==",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2018-07-23T10:20:35Z",
            "event": "mentioned",
            "id": 1747349955,
            "node_id": "MDE0Ok1lbnRpb25lZEV2ZW50MTc0NzM0OTk1NQ==",
            "url": "https://api.github.com/repos/cython/cython/issues/events/1747349955"
        },
        {
            "actor": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "node_id": "MDQ6VXNlcjQ5MTY1OQ==",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2018-07-23T10:20:35Z",
            "event": "subscribed",
            "id": 1747349956,
            "node_id": "MDE1OlN1YnNjcmliZWRFdmVudDE3NDczNDk5NTY=",
            "url": "https://api.github.com/repos/cython/cython/issues/events/1747349956"
        },
        {
            "actor": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "node_id": "MDQ6VXNlcjQ5MTY1OQ==",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2018-07-24T09:39:22Z",
            "event": "mentioned",
            "id": 1749696148,
            "node_id": "MDE0Ok1lbnRpb25lZEV2ZW50MTc0OTY5NjE0OA==",
            "url": "https://api.github.com/repos/cython/cython/issues/events/1749696148"
        },
        {
            "actor": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "node_id": "MDQ6VXNlcjQ5MTY1OQ==",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2018-07-24T09:39:22Z",
            "event": "subscribed",
            "id": 1749696150,
            "node_id": "MDE1OlN1YnNjcmliZWRFdmVudDE3NDk2OTYxNTA=",
            "url": "https://api.github.com/repos/cython/cython/issues/events/1749696150"
        }
    ],
    "events_url": "https://api.github.com/repos/cython/cython/issues/2505/events",
    "html_url": "https://github.com/cython/cython/issues/2505",
    "id": 342433559,
    "labels": [],
    "labels_url": "https://api.github.com/repos/cython/cython/issues/2505/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "MDU6SXNzdWUzNDI0MzM1NTk=",
    "number": 2505,
    "repository_url": "https://api.github.com/repos/cython/cython",
    "state": "open",
    "title": "Segfault when cythonizing",
    "updated_at": "2018-07-24T09:39:22Z",
    "url": "https://api.github.com/repos/cython/cython/issues/2505",
    "user": {
        "avatar_url": "https://avatars1.githubusercontent.com/u/506116?v=4",
        "events_url": "https://api.github.com/users/wkerzendorf/events{/privacy}",
        "followers_url": "https://api.github.com/users/wkerzendorf/followers",
        "following_url": "https://api.github.com/users/wkerzendorf/following{/other_user}",
        "gists_url": "https://api.github.com/users/wkerzendorf/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/wkerzendorf",
        "id": 506116,
        "login": "wkerzendorf",
        "node_id": "MDQ6VXNlcjUwNjExNg==",
        "organizations_url": "https://api.github.com/users/wkerzendorf/orgs",
        "received_events_url": "https://api.github.com/users/wkerzendorf/received_events",
        "repos_url": "https://api.github.com/users/wkerzendorf/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/wkerzendorf/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/wkerzendorf/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/wkerzendorf"
    }
}